independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -208,25 +212,80 @@ public void handle(Void event) {
                                 }
                                 requestContext.resume();
                             } else {
-                                clientResponse.bodyHandler(new Handler<>() {
-                                    @Override
-                                    public void handle(Buffer buffer) {
-                                        if (loggingScope != LoggingScope.NONE) {
-                                            clientLogger.logResponse(clientResponse, false);
-                                        }
-                                        try {
-                                            if (buffer.length() > 0) {
-                                                requestContext.setResponseEntityStream(
-                                                        new ByteArrayInputStream(buffer.getBytes()));
-                                            } else {
-                                                requestContext.setResponseEntityStream(null);
+                                if (requestContext.isFileDownload()) {
+                                    // when downloading a file we copy the bytes to the file system and manually set the entity type
+                                    // this is needed because large files can cause OOM or exceed the InputStream limit (of 2GB)
+
```

Comment:
it would be nice to log sth here too, e.g. reuse `logRequest(HttpClientRequest request, Buffer body, boolean multipart) ` but change `multipart` to `bodyOmitted` (and the log message accordingly)

Reply:
Will do